save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/codeeditor/CodeEditorComponent.kt

```diff
@@ -221,6 +227,49 @@ private fun codeEditorComponent() = FC<CodeEditorComponentProps> { props ->
             }
         }
 
-        aceBuilder(props.editorText, selectedMode, selectedTheme, props.selectedFile == null, props.onEditorTextUpdate)
+        aceBuilder(
+            props.editorText,
+            selectedMode,
+            selectedTheme,
+            getAceMarkers(props.savedEditorText, props.editorText),
+            props.selectedFile == null,
+            props.onEditorTextUpdate,
+        )
     }
 }
+
+private fun getAceMarkers(oldFile: String, newFile: String): Array<AceMarker> = getChangedLinesIndices(oldFile, newFile)
+    .map { getAceMarker(it) }
+    .toTypedArray()
+
+private fun getAceMarker(lineIndex: Int): AceMarker = jso {
+    startRow = lineIndex
+    endRow = lineIndex
+    startCol = 0
+    endCol = 1
+    className = "unsaved-marker"
+    type = "fullLine"
+    inFront = false
+}
+
+private fun getChangedLinesIndices(oldFile: String, newFile: String): Set<Int> {
```

Comment:
Any plans on integrating kotlin-multiplatform-diff here? 👉👈

Reply:
Will do it

Reply:
Done